### PR TITLE
Feature/rewrite in python

### PR DESCRIPTION
**Pull request by LiyouZhou**

### CODE REVIEW

**Comment by janjongboom** [comment]
Cool, I'll take a look at it tomorrow.

**Comment by janjongboom** [comment]
Could you add a section in README describing which flag to add to a profile, in case someone uses custom profile?

File: README.md [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
+```
 
-For performing the analysis, you need to recompile your program:
+Now run the ELF linker statistics tool:
 ```bash
-# Recompile project for chosen platform
-mbed compile -m K64F -t GCC_ARM -c
 # Process Data: provide one or more elf files for analysis
-php ../mbed-os-linker-report/elfsize.php ./.build/K64F/GCC_ARM/mbed-os-example-blinky.elf > ../mbed-os-linker-report/html/data-flare.js
+python ../mbed-os-linker-report/elfsize.py -i ./.build/K64F/GCC_ARM/mbed-os-example-blinky.elf -b
```
Comment (janjongboom) [inline README.md]: This needs to be `BUILD/K64F/GCC_ARM`. `.build` is old school mbed CLI.

File: README.md [hunk 2]
```diff
@@ -38,9 +39,9 @@ make -C mbed-os/features/FEATURE_UVISOR/importer
 # Recompile mbed-os app
 mbed compile -m K64F -t GCC_ARM -c
```
Comment (janjongboom) [inline README.md]: This should also include the profile information.

File: README.md [hunk 3]
```diff
@@ -38,9 +39,9 @@ make -C mbed-os/features/FEATURE_UVISOR/importer
...
 # Combine both elf outputs into a singe JSON file
-../mbed-os-linker-report/elfsize.php mbed-os/features/FEATURE_UVISOR/importer/TARGET_IGNORE/uvisor/platform/kinetis/release/configuration_kinetis_cortex_m4_0x1fff0000.elf .build/K64F/GCC_ARM/mbed-os-example-uvisor.elf > ../mbed-os-linker-report/html/data-flare.js
+python ../mbed-os-linker-report/elfsize.py -i mbed-os/features/FEATURE_UVISOR/importer/TARGET_IGNORE/uvisor/platform/kinetis/release/configuration_kinetis_cortex_m4_0x1fff0000.elf .build/K64F/GCC_ARM/mbed-os-example-uvisor.elf -b
```
Comment (janjongboom) [inline README.md]: `.build`

File: elfsize.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+from subprocess import check_output
+import re
+from os import path
+import os, json
+from collections import OrderedDict
+
+nm = "arm-none-eabi-nm"
```
Comment (janjongboom) [inline elfsize.py]: Needs a note that you need `arm-none-eabi-nm` in your path.

File: README.md (hunk 3 above)
Comment (janjongboom) [reply README.md]: This file is gone `mbed-os/features/FEATURE_UVISOR/importer/TARGET_IGNORE/uvisor/platform/kinetis/release/configuration_kinetis_cortex_m4_0x1fff0000.elf` so this instruction no longer works on uVisor. @meriac any idea what this should be now?

**Comment by janjongboom** [comment]
Tested for normal Mbed OS 5.6 application and that works nice. uVisor does not work, see comment in the review. /cc @meriac

**Comment by jupe** [comment]
could you “productionize” this e.g. adding setup.py.. I would like to integrate this to othet tools. Also unit/module tests would be more than welcome..

**Comment by LiyouZhou** [comment]
@janjongboom updated. I have tested the uvisor example. It works. I think the file is generated in the `make -C mbed-os/features/FEATURE_UVISOR/importer` step. The comment in the readme says that you might need to run make twice. I haven't found that. but might be needed for your setup. Also, I had to upgrade my arm-none-eabi-gcc for the makefile to successfully run through.

**Comment by LiyouZhou** [comment]
@jupe Good suggestions. could you write that into an issue so we don't lose them. That being said, I am just hacking around in my own time on this. Maybe a while before I get to these. Soooo, PR welcome :)

**Comment by janjongboom** [comment]
@LiyouZhou OK, Coolio. In that case LGTM from my side.

**Pull request merged**